tests/functional/test_install_upgrade.py

```diff
@@ -221,7 +221,7 @@ def test_uninstall_before_upgrade_from_url(script):
     )
     result2 = script.pip(
         'install',
-        'https://pypi.python.org/packages/source/I/INITools/INITools-'
+        'https://files.pythonhosted.org/packages/source/I/INITools/INITools-'
```

Comment:
What's the reference to pythonhosted.org? Why are we depending on files hosted there, rather than on PyPI like we did previously? Who will be maintaining those files (and who even has access to them)?

I'd prefer to continue getting our test files from PyPI, if we're not bundling them with the test suite.

Reply:
That's where Warehouse stores files, it doesn't store it on the same domain anymore because there is security risk in the browser. You'll see it common amongst sites that allow users to upload content to have two domains, the website's domain and the "user content" domain. For Warehouse, ``files.pythonhosted.org`` is the user content domain.

You can see Github announcing a similar change at https://developer.github.com/changes/2014-04-25-user-content-security/.

So this is still getting our test files from PyPI :)

Reply:
Interesting. Is that documented anywhere? Presumably if we moved our hosting domain, that would break all our references - whereas in the old tests, we'd have to rename PyPI before our tests would break.

Ultimately, if the tests work, it's not a big deal right now, but it does have an impact on the maintainability of our tests in the long run. So while I'm OK with it as an expedient way to get the tests working in the context of this PR, I'd like a proper discussion longer term (maybe even relocating these files into the test directory - I'd rather we didn't have a network dependency we don't need).

Reply:
Yea, we desperately need to break the reliance on a network dependency in our test suite :(

I'm not sure if we ever explicitly documented that change TBH, it's been like that for like 2 years in Warehouse (it's one of the very first bits of Warehouse I completed) so I don't really remember what was documented about it (if anything).

Reply:
Heh, it was mostly a rhetorical question. I'd have been surprised if we had documented it, my impression of pythonhosted.org was that it was a mostly-obsolete documentation site, TBH.

Reply:
Yea, I ended up re-using that domain because it was there and having ``pythonhosted.org`` be the place for stuff that's hosted by Python, but isn't produced by Python seemed reasonable. The docs are at ``https://pythonhosted.org/`` while the files are at ``https://files.pythonhosted.org/``.